### PR TITLE
first look on refactoring

### PR DESCRIPTION
**Pull request by darkdef**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ✔️
| Fixed issues  |

### CODE REVIEW

File: src/MessageSource.php [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            $needUpdate = false;
+            if (isset($translateMessages[$message_id]) && $translateMessages[$message_id] !== $translation) {
+                $this->db->createCommand()->delete($this->messageTable, ['id' => $sourceMessages[$message_id]])->execute();
+                $needUpdate = true;
+            }
+
+            if ($needUpdate || !isset($translateMessages[$message_id])) {
+                $result = $this->db->getSchema()->insert($this->messageTable, ['id' => $sourceMessages[$message_id], 'locale' => $locale, 'translation' => $translation]);
+                if ($result===false)
+                    throw new \RuntimeException('Can not create source message with id ' . $message_id);
```
Comment (samdark) [inline src/MessageSource.php:92]:
```suggestion
                if ($result === false) {
                    throw new \RuntimeException('Can not create source message with id ' . $message_id);
                }
```

File: src/MessageSource.php [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    {
+        $sourceMessages = (new Query($this->db))
+            ->select(['id', 'message_id'])
+            ->from($this->sourceMessageTable)
+            ->where(['category' => $category])
+            ->all();
+
+        $sourceMessages = ArrayHelper::map($sourceMessages, 'message_id', 'id');
+
+        $translateMessages = $this->read($category, $locale);
```
Comment (samdark) [inline src/MessageSource.php:73]:
```suggestion
        $translatedMessages = $this->read($category, $locale);
```

File: src/MessageSource.php [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+        foreach ($messages as $message_id => $translation) {
```
Comment (samdark) [inline src/MessageSource.php:75]:
```suggestion
        foreach ($messages as $messageId => $translation) {
```

File: src/MessageSource.php [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+            if (!isset($sourceMessages[$message_id])) {
+                $result = $this->db->getSchema()->insert($this->sourceMessageTable, ['category' => $category, 'message_id' => $message_id]);
+                if ($result===false)
+                    throw new \RuntimeException('Can not create source message with id ' . $message_id);
```
Comment (samdark) [inline src/MessageSource.php:79]:
```suggestion
                if ($result === false) {
                    throw new \RuntimeException('Can not create source message with id ' . $message_id);
                }
```

**Review by samdark — APPROVED** [review]
👍 but please fix minor issues.

File: src/MessageSource.php [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+                ])
+            ->where([
+                'locale' => $locale,
+            ]);
+        $messages = $query->all();
+
+        return ArrayHelper::map($messages, 'message_id', 'translation');
+    }
+
+    public function setSourceMessageTable(string $sourceMessageTable): void
```
Comment (samdark) [inline src/MessageSource.php:77]: Do we need to change it after creating the message source?

File: src/MessageSource.php [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+    {
+        $this->sourceMessageTable = $sourceMessageTable;
+    }
+
+    public function setMessageTable(string $messageTable): void
```
Comment (samdark) [inline src/MessageSource.php:82]: Do we need to change it after creating the message source?

File: src/MessageSource.php [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+
+            if ($needUpdate || !isset($translatedMessages[$messageId])) {
+                $result = $this->db->getSchema()->insert($this->messageTable, ['id' => $sourceMessages[$messageId], 'locale' => $locale, 'translation' => $translation]);
+                if ($result === false)
+                    throw new \RuntimeException('Can not create source message with id ' . $messageId);
+            }
+        }
+    }
+
+    public function getCacheKey(string $category, string $locale): string
```
Comment (samdark) [inline src/MessageSource.php:121]: Should it be public? Why do we need it externally?

File: src/MessageSource.php (hunk 6 above)
Comment (darkdef) [reply src/MessageSource.php]: That's not needed

File: src/MessageSource.php (hunk 7 above)
Comment (darkdef) [reply src/MessageSource.php]: that's error, fixed

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**